src/Symfony/Component/Cache/Marshaller/SodiumMarshaller.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Marshaller;
+
+use Symfony\Component\Cache\Exception\CacheException;
+
+/**
+ * Encrypt/decrypt values using Libsodium.
+ *
+ * @author Ahmed TAILOULOUTE <ahmed.tailouloute@gmail.com>
+ */
+class SodiumMarshaller implements MarshallerInterface
+{
+    private $marshaller;
+    private $decryptionKey;
+
+    public function __construct(string $decryptionKey, MarshallerInterface $marshaller = null)
+    {
+        if (!self::isSupported()) {
+            throw new CacheException('The "sodium" PHP extension is not loaded.');
+        }
+
+        $this->marshaller = $marshaller ?? new DefaultMarshaller();
+        $this->decryptionKey = $decryptionKey;
+    }
+
+    public static function isSupported(): bool
+    {
+        return \function_exists('sodium_crypto_box_seal');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function marshall(array $values, ?array &$failed): array
+    {
+        return array_map(function ($value) {
+            return sodium_crypto_box_seal($value, sodium_crypto_box_publickey($this->decryptionKey));
```

Comment:
The key could be extracted only once

Reply:
Done, 7aee40ce0f73d54b23010383d65b6c26fba03c91